src/packageurl/__init__.py

```diff
@@ -285,16 +285,21 @@ def __str__(self, *args, **kwargs):
     def __hash__(self):
         return hash(self.to_string())
 
-    def to_dict(self, encode=False):
+    def to_dict(self, encode=False, empty=None):
         """
-        Return an ordered dict of purl components as {key: value}. If `encode`
-        is True, then "qualifiers" are encoded as a normalized string.
-        Otherwise, qualifiers is a mapping.
+        Return an ordered dict of purl components as {key: value}.
+        If `encode` is True, then "qualifiers" are encoded as a normalized
+        string. Otherwise, qualifiers is a mapping.
+        You can provide a value for `empty` to be used in place of default None.
         """
         data = self._asdict()
         if encode:
             data['qualifiers'] = normalize_qualifiers(self.qualifiers,
                                                       encode=encode)
+
+        if empty is not None:
+            data = OrderedDict((k, v or empty) for k, v in data.items())
```

Comment:
Since data is already an OrderedDict... you may want to modify it in place but this is minor:
```
for k, v in data.items():
    data[k] = v or empty
```

Reply:
Changed in https://github.com/package-url/packageurl-python/pull/40/commits/f2e962fe97801cfe2ecfdc6f6cec9850aced93df